src/multicharge/model/eeqbc.f90

```diff
@@ -48,729 +63,900 @@ module multicharge_model_eeqbc
       real(wp) :: kbc
       !> Exponent of the distance/CN normalization
       real(wp) :: norm_exp
+      !> vdW radii
+      real(wp), allocatable :: rvdw(:, :)
    contains
-      !> Update multicharge cache
       procedure :: update
+      procedure :: get_coulomb_matrix
+      procedure :: get_coulomb_derivs
       !> Calculate right-hand side (electronegativity)
-      procedure :: get_vrhs
-      !> Calculate Coulomb matrix 
+      procedure :: get_xvec
+      procedure :: get_xvec_derivs
+      !> Calculate Coulomb matrix
       procedure :: get_amat_0d
       !> Calculate Coulomb matrix periodic
       procedure :: get_amat_3d
       !> Calculate Coulomb matrix derivative
       procedure :: get_damat_0d
       !> Calculate Coulomb matrix derivative periodic
       procedure :: get_damat_3d
+      !> Calculate constraint matrix
+      procedure :: get_cmat_0d
+      procedure :: get_cmat_diag_3d
+      procedure :: get_dcmat_0d
+      ! procedure :: get_dcmat_3d
```

Comment:
why would we not need these subroutines?

Reply:
Because the dcmat_3d will likely be calculated again together with the Coulomb matrix derivative. But this is not very important. 

The doxygen comments with the purpose of each function would still be good